tests/unittests/TagServiceTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace OCA\GDataVaas\Tests;
+
+use OCA\GDataVaas\Db\DbFileMapper;
+use OCA\GDataVaas\Service\TagService;
+use OCP\SystemTag\ISystemTag;
+use OCP\SystemTag\ISystemTagManager;
+use OCP\SystemTag\ISystemTagObjectMapper;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\Test\TestLogger;
+
+class TagServiceTest extends TestCase {
+	public static $OBJECT_ID_1 = 1;
+
+	private function getTagManager(): ISystemTagManager {
+		$tagManager = $this->createMock(ISystemTagManager::class);
+		$tagManager->method('getTag')->willReturnCallback(function ($name, $create) {
+			$tag = $this->createMock(ISystemTag::class);
+			$tag->method('getId')->willReturn($name);
+			return $tag;
+		});
+		return $tagManager;
+	}
+
+	public function testSetWontScan_ShouldDeleteCleanAndMaliciousAndAssignWontScan() {
+		$tagManager = $this->getTagManager();
+
+		$tagMapper = $this->createMock(ISystemTagObjectMapper::class);
+		$tagMapper->method('getTagIdsForObjects')->willReturn([self::$OBJECT_ID_1 => [TagService::CLEAN, TagService::MALICIOUS]]);
+		$unassignTagsMatcher = $this->exactly(2);
+		$tagMapper->expects($unassignTagsMatcher)->method('unassignTags')->willReturnCallback(function ($objectId, $objectType, $tagIds) use ($unassignTagsMatcher) {
+			switch($unassignTagsMatcher->numberOfInvocations()) {
+				case 1:
+					$this->assertEquals($objectId, strval(self::$OBJECT_ID_1));
+					$this->assertEquals($objectType, 'files');
+					$this->assertEquals($tagIds, [TagService::CLEAN]);
```

Comment:
The assertions are misordered. The actual value should be second.
![Screenshot 2024-07-11 093658](https://github.com/GDATASoftwareAG/nextcloud-gdata-antivirus/assets/99874374/6da27d5e-d53c-432b-862d-dd0b0319f354)